File: programmers/lev2/12977.py
# 그냥 눈에 보이는대로 첫번째 시도
from itertools import combinations
import math

def solution(nums):
    answer = 0


    test_data=list(map(sum,list(combinations(nums,3))))
    print(test_data)
    for sum_data in test_data:
        sqrt=int(math.sqrt(sum_data))
        for n in range(2,sqrt+1):
            if sum_data%n==0:
                break
            if n==sqrt and sum_data%n!=0:
                answer+=1
    print(answer)
    return answer

nums=[1,2,7,6,4]
solution(nums)

File: java/src/programmers/year2023/lv1/_12977.java
package programmers.year2023.lv1;

public class _12977 {
    public static void main(String[] args) {
//        int[] nums = {1, 2, 3, 4};
        int[] nums = {1, 2, 7, 6, 4};
        solution(nums);
    }

    public static int solution(int[] nums) {
        int answer = 0;
        int size = nums.length;
        int firstIdx = 0;
        int secondIdx = 1;
        int thirdIdx = 2;

        while (true) {
            if (firstIdx == size - 2) {
                break;
            }

            if (secondIdx == size - 1) {
                firstIdx++;
                secondIdx = firstIdx + 1;
                thirdIdx = firstIdx + 2;
                continue;
            }


            if (thirdIdx == size) {
                secondIdx++;
                thirdIdx = secondIdx + 1;
                continue;
            }

            int sum = nums[firstIdx] + nums[secondIdx] + nums[thirdIdx];
            if (isPrime(sum)) {
                answer++;
            }


            thirdIdx++;

            // ***11 **1*1 **11* *1**1 *1*1* *11** 11111 11111
        }

        return answer;
    }

    private static boolean isPrime(int num) {
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
}
